sklearn/covariance/_robust_covariance.py

```diff
@@ -121,10 +121,9 @@ def _c_step(
     dist = np.inf
 
     # Initialisation
-    support = np.zeros(n_samples, dtype=bool)
     if initial_estimates is None:
         # compute initial robust estimates from a random subset
-        support[random_state.permutation(n_samples)[:n_support]] = True
+        support = random_state.permutation(n_samples)[:n_support]
```

Comment:
Maybe `support` should be renamed to `support_indices` when ever it contains integer indices of data points and to `support_mask` whenever it is a boolean mask.

Uses is the same variable names for both kinds of data structures is a bit confusing.